apps/daimo-mobile/src/view/shared/SwipeUpDown.tsx

```diff
@@ -81,22 +80,32 @@ export const SwipeUpDown = forwardRef<SwipeUpDownRef, SwipeUpDownProps>(
     });
 
     const [isMini, setIsMini] = useState(true);
+    const historyOpOpacity = useSharedValue(0);
 
     // When user selects a transaction, open the bottom sheet part way.
     const animatedIndex = useSharedValue(0);
     const sheetExpand = () => {
       if (animatedIndex.value === 1) {
         bottomRef.current?.snapToIndex(2);
       }
-      setSnapPoints([posYMini, 450, posYFull]);
-      if (animatedIndex.value === 0) {
-        bottomRef.current?.snapToIndex(1);
-      }
+      snapPoints.value = [posYMini, 450, posYFull];
+      // skip one frame to let snapPoints addapt
+      // without it it works perfectly fine
+      // but the backdrop is gray right away without smooth transition
+      setTimeout(() => {
+        if (animatedIndex.value === 0) {
+          bottomRef.current?.snapToIndex(1);
+          historyOpOpacity.value = 1;
```

Comment:
what's the purpose of this line? 
it looks like we're overwriting to `withTiming(1)` immediately after

Reply:
When I call `historyOpOpacity.value = 1;` just before `withTiming(1)` I basically skip the animation of the withTiming because the value is already there.

The `historyOpOpacity.value = 1;` is inside `if` because I wanted to skip the animation conditionally but now when I look at it maybe I will use more undestandable syntax:
```
historyOpOpacity.value = animatedIndex.value === 0 ? 1 : withTiming(1)
```